src/frontend/tsx/GamePanel.tsx

```diff
@@ -24,46 +28,66 @@ export function GamePanel () {
     await pokeAction(urbit, claimSpecialDraw(gameID))
   }
 
+  const moveOpacity = (index: number) => {
+    if (displayIndex == null || index <= displayIndex) {
+      return 1.0
+    } else {
+      return 0.3
+    }
+  }
+
+  const moveList = () => {
+    let displayMoves = (displayGame.info.moves !== null) ? displayGame.info.moves : []
+    let components = []
+    for (let wIndex: number = 0; wIndex < displayMoves.length; wIndex += 2) {
+      const move: number = (wIndex / 2) + 1
+      const bIndex: number = wIndex + 1
+      const wMove: SAN = displayMoves[wIndex].san
+
+      if (bIndex >= displayMoves.length) {
+        components.push(
+          <li key={ move } className='move-item' style={{ opacity: moveOpacity(wIndex) }}>
+            <span onClick={ () => setDisplayIndex(wIndex) }>
+              { wMove }
+            </span>
+          </li>
+        )
+      } else {
+        components.push(
+          <li key={ move } className='move-item' style={{ opacity: moveOpacity(wIndex) }}>
+            <span onClick={ () => setDisplayIndex(wIndex) }>
+              { wMove }
+            </span>
+            { '\xa0'.repeat(6 - wMove.length) }
+            <span onClick={ () => setDisplayIndex(bIndex) } style={{ opacity: (moveOpacity(wIndex) == 0.3) ? 1.0 : moveOpacity(bIndex) }}>
```

Comment:
We should add a comment here that opacity ratings are cumulative, because it took me some time to understand why we were flipping the values here.

Duplicating specific hard values like this is an anti-pattern. Consider the case where we decide we want to change the opacity to `0.4` instead: now we have to change a specific value in two places in the code. The two common solutions are: 1) give the value a name, such as `moveOpacity`, and use the name instead, so that the value is only assigned once; 2) Change the way the value is assigned so that a single location in a function assigns it.

For the above reason, I would recommend swapping the ternary operator condition and arms:
```
opacity: (moveOpacity(wIndex) == 1.0) ? moveOpactiy(bIndex) : 1.0
```
This is more acceptable, since 100% is a natural default opacity (though if we ever wanted to change that, we'd now need named constants or a new `moveOpacity` function).

Reply:
Ah yeah, this is much more elegant.